internal/overlord/servstate/handlers.go

```diff
@@ -75,9 +75,17 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 		return fmt.Errorf("cannot find service %q in plan", req.Name)
 	}
 
-	_, previous := m.services[req.Name]
-	if previous {
-		return fmt.Errorf("service %q was previously started", req.Name)
+	if service, previous := m.services[req.Name]; previous {
+		select {
+		case <-service.done:
+			// Service exited but not yet removed, safe to override.
+		default:
+			// Already started
+			m.state.Lock()
+			task.Logf("service %q already started", req.Name)
```

Comment:
The error message became a log, so we need to capitalize and add punctuation so it fits in the context:
```
"Service %q already started."
```